File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char> st;
        int n=s.size();
        for(int i=0;i<n;i++){
            if(!st.empty() && ((st.top() == 'A' && s[i] == 'B') ||
                (st.top() =='C' && s[i]=='D'))){
                st.pop();
            }else{
                st.push(s[i]);
            }
        }
        return st.size();
    }
};

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings .py
class Solution:
    def minLength(self, s: str) -> int:
        s1=[]
        for p in s:
            if(s1 and((s1[-1]=="A" and p=="B") or(s1[-1]=="C" and p=="D"))):
                s1.pop()
            else:
                s1.append(p)
        return (len(s1))